### PR TITLE
fix: remove line break from empty object representation in resolved c…

### LINKED ISSUE #21790: In resolved config, do not show empty object with line break
image (14) (1)

┆Issue is synchronized with this [Jira Task](https://cypress-io.atlassian.net/browse/UNIFY-1463) by [Unito](https://www.unito.io)
┆Attachments: <a href="https://cypress-io.atlassian.net/rest/api/2/attachment/content/11656">image (14).png</a>
┆author: Alejandro Estrada
┆epic: UI/UX Improvements
┆friendlyId: UNIFY-1463
┆priority: Low
┆sprint: Backlog
┆taskType: Task

### PR DESCRIPTION
**Pull request by Pashozator**
- Closes https://github.com/cypress-io/cypress/issues/21790

### User facing changelog

Don't render `<br/>` tag when object is empty

### How has the user experience changed?

Before
before-empty

After
after-empty

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Comment by mike-plummer** [comment]
Hi @Pashozator , thanks for opening a PR! Could you fill in the PR template please? It looks like the majority of it was accidentally removed, I've updated the PR description to add the missing sections & checklist back in

File: packages/app/src/settings/project/renderers/RenderObject.vue [hunk 1]
```diff
@@ -46,10 +46,9 @@
     <span
       :class="props.colorClasses"
       :data-cy-config="props.from"
-    >{</span><br><template
-      v-for="(val, k) in props.value"
-      :key="k"
-    >
+    >{</span>
+    <br v-if="!Object.keys(props.value).length">
+    <template v-for="(val, k) in props.value" :key="k">
```
Comment (mike-plummer) [inline packages/app/src/settings/project/renderers/RenderObject.vue]: `props.value` isn't always an `object` so this logic is breaking how non-empty sections are rendering in some situations. You can see the Typescript type for `value` in the prop definitions `Record<string, any> | any[]`

**Before**
Screen Shot 2022-12-08 at 1 26 49 PM

**After**
Screen Shot 2022-12-08 at 1 26 40 PM

**Review by mike-plummer — CHANGES_REQUESTED** [review]

**Comment by emilyrohrbough** [comment]
@Pashozator Have you had a chance to look into the feedback @mike-plummer left?

**Comment by Pashozator** [comment]
@emilyrohrbough not yet, still I have some problems with installing dependencies in repo.

```
Internal Error: Duplicate workspace name cypress: /Users/pashozator/cypress/cli conflicts with /Users/pashozator/cypress
    at He.addWorkspace (/Users/pashozator/.cache/node/corepack/yarn/3.3.0/yarn.js:439:1920)
    at async He.setupWorkspaces (/Users/pashozator/.cache/node/corepack/yarn/3.3.0/yarn.js:439:1614)
    at async He.find (/Users/pashozator/.cache/node/corepack/yarn/3.3.0/yarn.js:436:1559)
    at async Ou.execute (/Users/pashozator/.cache/node/corepack/yarn/3.3.0/yarn.js:499:12005)
    at async Ou.validateAndExecute (/Users/pashozator/.cache/node/corepack/yarn/3.3.0/yarn.js:63:13148)
    at async TA.run (/Users/pashozator/.cache/node/corepack/yarn/3.3.0/yarn.js:77:2354)
    at async TA.runExit (/Users/pashozator/.cache/node/corepack/yarn/3.3.0/yarn.js:77:2538)
    at async i (/Users/pashozator/.cache/node/corepack/yarn/3.3.0/yarn.js:446:12061)
    at async t (/Users/pashozator/.cache/node/corepack/yarn/3.3.0/yarn.js:446:10278)
pashozator@MacBook-Air-pashozator cypress % 
```

**Comment by lmiller1990** [comment]
We manage the Cypress monorepo with Yarn 1.x - it looks like you have Yarn 3 installed globally: `yarn/3.3.0`. Try using Yarn 1.x to install.

Once this is ready to go, can you post a screenshot of before/after? I'd recommend using Component Testing and opening `ConfigCode.cy.tsx` in `packages/app` - it's got a lot of different scenarios.

**Comment by Pashozator** [comment]
> We manage the Cypress monorepo with Yarn 1.x - it looks like you have Yarn 3 installed globally: `yarn/3.3.0`. Try using Yarn 1.x to install.
> 
> Once this is ready to go, can you post a screenshot of before/after? I'd recommend using Component Testing and opening `ConfigCode.cy.tsx` in `packages/app` - it's got a lot of different scenarios.

lovely stuff, thanks will update Pull Request in few hours

**Comment by lmiller1990** [comment]
@Pashozator looks like you've got a failing test still: https://app.circleci.com/pipelines/github/cypress-io/cypress/47843/workflows/cace73c8-df54-411b-b029-3cdcaec30298/jobs/1998781.

`cd packages/app && yarn cypress:open` locally to work on it. You can run the suite with `yarn cypress:run:ct --browser chrome` locally.

Also please include before/after screenshot, that will help w/ reviewing greatly. Thank you!

**Comment by Pashozator** [comment]
> Also please include before/after screenshot, that will help w/ reviewing greatly. Thank you!

@lmiller1990 I added screenshots in PR description
@lmiller1990 if you can please rerun tests, seems like they are flacky - failures not related to my change imo

File: packages/app/src/settings/project/renderers/RenderObject.vue [hunk 2]
```diff
@@ -46,7 +46,10 @@
     <span
       :class="props.colorClasses"
       :data-cy-config="props.from"
-    >{</span><br><template
+    >{</span>
+    <!-- Do not render break line for empty object or array -->
+    <br v-if="Array.isArray(props.value) ? props.value.length : Object.keys(props.value).length">
```
Comment (emilyrohrbough) [inline packages/app/src/settings/project/renderers/RenderObject.vue]: will `Array.isArray(props.value)` ever be hit? It looks like the first template is used above when `<template v-if="Array.isArray(value)">`

Comment (Pashozator) [reply packages/app/src/settings/project/renderers/RenderObject.vue]: It should not but https://github.com/cypress-io/cypress/pull/25053#discussion_r1043746922 misleaded me.

Comment (emilyrohrbough) [reply packages/app/src/settings/project/renderers/RenderObject.vue]: ohh @mike-plummer do you think this is needed?

Comment (mike-plummer) [reply packages/app/src/settings/project/renderers/RenderObject.vue]: @emilyrohrbough Yeah, should be safe to remove. I didn't originally see that we had a separate template section for when `value` is an array

Comment (Pashozator) [reply packages/app/src/settings/project/renderers/RenderObject.vue]: @emilyrohrbough done

**Review by emilyrohrbough — APPROVED** [review]

**Review by mike-plummer — APPROVED** [review]

**Comment by emilyrohrbough** [comment]
Thank you @Pashozator for fixing this!!

**Pull request merged**